File: mi_conjunto_guardas/lib/models/package.dart
import 'company.dart';

/// Modelo de 'encomienda'
class Package {
  int packageId;
  String receptionDate;
  String deliveryDate;
  String notes;
  bool checked;
  Company company;

  Package(
      {this.packageId,
      this.receptionDate,
      this.deliveryDate,
      this.notes,
      this.checked,
      this.company});

  factory Package.fromJson(Map<String, dynamic> json) {
    return Package(
        packageId: json["packageId"],
        receptionDate: json["receptionDate"],
        deliveryDate: json["deliveryDate"],
        notes: json["notes"],
        checked: false,
        company: Company.fromJson(json["Company"]));
  }
}

File: server/sequelize/models/Package.js
import DataTypes from "sequelize";

/**
 * Modelo de 'encomienda' en base de datos
 * @param {*} sequelize 
 */
export const packageModel = (sequelize) => {
  sequelize.define(
    "Package",
    {
      packageId: {
        type: DataTypes.INTEGER,
        primaryKey: true,
        allowNull: false,
        autoIncrement: true,
        field: "id_encomienda",
      },
      receptionDate: {
        type: DataTypes.DATE,
        allowNull: false,
        field: "fecha_recepcion",
      },
      deliveryDate: {
        type: DataTypes.DATE,
        field: "fecha_entrega",
      },
      notes: {
        type: DataTypes.STRING,
        field: "observaciones",
      },
    },
    {
      tableName: "encomienda",
      timestamps: false,
    }
  );
};
